File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s=0
        max_s=-1000000
        for i in nums:
            t=s+i
            s=max(0,s+i)
            if (t>max_s):
                max_s=t
        return max_s

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) 
    {
        let max=0;
        let l=0;
        let r= heights.length-1;
        while (l<r){
            let mini= heights[l]<heights[r] ? l : r;
            max= max<(heights[mini]*(r-l)) ? (heights[mini]*(r-l)) : max;
            if (mini==l){
                l+=1;
            }
            else{
                r-=1;
            }
        }
        return max;
    }
}
